### DISCUSSION TITLE
Dash in module name makes reuse harder than necessary

### DISCUSSION THREAD

**Discussion by chrisv2** [O-X-L/ansible-webui-v0#56] [General]
I'd like to integrate some functionality of this nice project into my own Django app, unfortunately the top level module has a dash in it, so it is not easily importable:

```python
# does not work => syntax error!
from ansibleguy-webui import aw

# this works, but breaks static analysis (e.g. IDE autocomplete)
import importlib
aw = importlib.import_module("ansibleguy-webui.aw")
```
Would you be open to a pull request changing the module from `ansibleguy-webui` to `ansibleguy_webui`?

**Comment by chrisv2** [comment]
Note: Turns out this is easier than I thought, see my branch [here](https://github.com/greybyte/webui/tree/rename_module)

**Comment by ansibleguy** [comment]
This is the name of the module on PIP this would break all existing installs.. :/
I've gone with this naming scheme as it is commonly used by the Ansible toolset.

**Comment by chrisv2** [comment]
Thank you very much for looking at my suggestion! The package name `ansibleguy-webui` is totally fine and does not need to be changed, it is only the name of the module which causes import issues. In other words, one would still do `pip install ansibleguy-webui`, only the run command would change (`python -m ansibleguy_webui`).

Making the top-level module importable would also allow to get rid of the [sys.path hackery](https://github.com/search?q=repo%3Aansibleguy%2Fwebui%20sys_path.append&type=code) which is currently necessary to import `aw`, the actual Django app.

**Comment by superstes** [comment]
If you want it - you could provide a PR that adds another pypi action  https://github.com/ansibleguy/webui/actions/workflows/release.yml where we additionally release the module under another name (`ansibleguy_webui_plugin` or something like that)

**Comment by ansibleguy** [comment]
The project is moving to a new repository and pip-module.
The new one uses underscores: https://github.com/O-X-L/ansible-webui
